### ISSUE TITLE
Inconvinient button addons with 3.1

### ISSUE THREAD

**Issue by nrother** [Holt59/cakephp3-bootstrap-helpers#108]
With the new template system used in 3.1 it's very inconvenient to add [button addons](http://getbootstrap.com/components/#input-groups-buttons) to a input.

Before:
```php
echo $this->Form->control('email', [
    'append' => $this->Form->button('text', [
        'type' => 'button'
    ])
]);
```
After
```php
$button = '<span class="input-group-btn">' . $this->Form->button('text', [
    'type' => 'button'
]) . '</span>';
echo $this->Form->control('email', [
    'templates' => [
        'input' => $this->Html->div('input-group', '<input type="{{type}}" name="{{name}}" class="form-control{{attrs.class}}" {{attrs}} />' . $button),
    ]
]);
```

Maybe the `append` and `prepend` options could be added again?

**Comment by Holt59** [comment]
The behavior of the `append` and `prepend`  options should not have been modified. I have unit tests on these that are ok so this is very strange... I will try to reproduce when I get back to a complete CakePHP installation.

**Comment by Holt59** [comment]
Bug found - In v3.1.0, only the `input` method is override, so if you call `control()`, you will call the parent's method, which does not have `prepend` or `append` options.

**Quick fix:** Use `input()` or try the `dev-v3.1.2` branch which overrides `control()`.

I will push a fix when I get a correct working environment!

**Comment by Holt59** [comment]
I am not sure I will fix this actually:

- The 3.1.0 and 3.1.0 versions will support CakePHP >= 3.2.3 and < 3.4.0. 
- The v3.1.2 branch currently requires (and thus support) CakePHP >= 3.4.0, so you should use it if you use CakePHP 3.4.0.

Let me know if you think this should really be fixed in 3.1.0 for some reasons.

**Comment by Holt59** [comment]
I have updated the requirements for v3.1.0 and v3.1.1 (see the README), and the v3.1.2 is now on the master branch, so I am closing this.
